File: app/src/debug/java/com/handstandsam/shoppingapp/debug/DebugPreferences.java
package com.handstandsam.shoppingapp.debug;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class DebugPreferences {

    public static final String MOCK_MODE = "mock_mode";
    public static final String CHUCK = "chuck";

    private final SharedPreferences sharedPreferences;

    public DebugPreferences(Context context) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
    }

    public boolean isMockMode() {
        boolean isMock = sharedPreferences.getBoolean(MOCK_MODE, false);
        return isMock;
    }

    public void setMockMode(boolean on) {
        sharedPreferences.edit().putBoolean(MOCK_MODE, on).apply();
    }

    public boolean isChuckEnabled() {
        boolean enabled = sharedPreferences.getBoolean(CHUCK, false);
        return enabled;
    }

    public void setChuckEnabled(boolean enabled) {
        sharedPreferences.edit().putBoolean(CHUCK, enabled).apply();
    }

}

File: debug/src/main/java/com/handstandsam/shoppingapp/debug/DebugPreferences.kt
package com.handstandsam.shoppingapp.debug

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class DebugPreferences(context: Context) {

    private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context.applicationContext)


    var isMockMode: Boolean
        get() = sharedPreferences.getBoolean(MOCK_MODE, false)
        set(on) = sharedPreferences.edit().putBoolean(MOCK_MODE, on).apply()

    var isChuckEnabled: Boolean
        get() = sharedPreferences.getBoolean(CHUCK, false)
        set(enabled) = sharedPreferences.edit().putBoolean(CHUCK, enabled).apply()

    companion object {
        val MOCK_MODE = "mock_mode"
        val CHUCK = "chuck"
    }

}
